configure.py

```diff
@@ -1910,7 +1910,7 @@ def query_seastar_flags(pc_file, link_static_cxx=False):
                 if binary not in tests_not_using_seastar_test_framework:
                     local_libs += ' ' + "$seastar_testing_libs_{}".format(mode)
                 else:
-                    local_libs += ' ' + '-lgnutls'
+                    local_libs += ' ' + '-lgnutls' + ' ' + '-lboost_unit_test_framework'
```

Comment:
Add matching change to CMakeLists.txt? Maybe @tchaikov will add it in a follow-up.
Does this need any new package installed (`install-dependencies.sh`)? I guess we will soon find out (CI).

Reply:
seastar's `install-dependencies.sh` installs much more than we actually need: see 

- https://github.com/scylladb/seastar/blob/7cbdfe4c42d845d030c5734c305212a1c5fa0637/install-dependencies.sh#L93
- https://github.com/scylladb/seastar/blob/7cbdfe4c42d845d030c5734c305212a1c5fa0637/install-dependencies.sh#L103
- https://github.com/scylladb/seastar/blob/7cbdfe4c42d845d030c5734c305212a1c5fa0637/install-dependencies.sh#L37

probably we should install the dependencies in a more finer grained manner. but anyway, we should be safe.

Reply:
> Does this need any new package installed (`install-dependencies.sh`)?

No we don't. The very same lib is linked into pure-boost-tests few lines up in configure.py